### PR TITLE
Add scaled(toMaxSize:NSSize) to NSImage, including a test

### PR DESCRIPTION
**Pull request by buddax2**
Add a function to NSImage to be able to scale an image to some maximum size. It could be useful for creating thumbnails.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.

### CODE REVIEW

File: Sources/Extensions/AppKit/NSImageExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import AppKit
+
+// MARK: - Methods
+extension NSImage {
+    
+    /// SwifterSwift: NSImage scaled to maximum size with respect to aspect ratio
+    ///
+    /// - Parameter toMaxSize: maximum size
+    /// - Returns: scaled NSImage
+    public func scaled(toMaxSize: NSSize) -> NSImage {
```
Comment (SD10) [inline Sources/Extensions/AppKit/NSImageExtensions.swift]: `public func scaled(to maxSize: NSSize) -> NSImage` may be better?

Comment (SD10) [reply Sources/Extensions/AppKit/NSImageExtensions.swift]: actually maybe not, loses the expressiveness of it being a max size.

**Comment by SD10** [comment]
Thank you for your contribution @buddax2. Can you add a `CHANGELOG` entry for this?

**Comment by buddax2** [comment]
@SD10 sure I can. Should I add a new release section for this (v4.0.2)?

**Comment by SD10** [comment]
No @buddax2 It just goes under the `Next Release` section. Follow the previous CHANGELOG as a guide

**Comment by buddax2** [comment]
@SD10 can I add an entry for #296 in this PR? :)

**Comment by SD10** [comment]
Sure things @buddax2 👍

**Comment by buddax2** [comment]
@SD10 done

**Comment by omaralbeik** [comment]
Thank you @buddax2 @SD10 👍

**Pull request merged**